### ISSUE TITLE
Test and train using adjacent points

### ISSUE THREAD

**Issue by palak-purohit** [UMDataScienceLab/SGD-in-Gaussain-processes#1]
We are working on reproducing the work "Stochastic Gradient Descent in Correlated Settings" for ReScience. According to the paper, "For borehole, otl circuit and wing weight datasets, we first learn the hyperparameters and then estimate the RMSE by predicting 40,000 adjacent test points using 60,000 nearby training points due to memory limit." 
How does one define adjacent and nearby in multidimensional case since these mathematical functions take in multiple inputs? What exactly is the input given to the functions to generate this data?

**Comment by UMDataScienceLab** [comment]
You may construct a k-d tree to conduct a nearest neighbor search, which finds the m − 1 nearest neighbor for every data point in a given dataset of size n in O(n log n) time and O(n) space

**Comment by patel-zeel** [comment]
Thank you for resolving the query @UMDataScienceLab. The above approach seems computationally expensive. Can you please give us some estimate on the practical test time for `m` test points and `n` train points for easy comparison?
